tools/line_endings.py

```diff
@@ -1,37 +1,47 @@
+#!/usr/bin/env python
 from __future__ import print_function
-import sys, shutil, os
+import os
+import sys
+
 
 def convert_line_endings(text, from_eol, to_eol):
-  if from_eol == to_eol: return text
+  if from_eol == to_eol:
+    return text
   return text.replace(from_eol, to_eol)
 
-# if dst_filename is empty, operates in place on src_filename
+
 def convert_line_endings_in_file(filename, from_eol, to_eol):
-  if from_eol == to_eol: return # No conversion needed
+  if from_eol == to_eol:
+    return # No conversion needed
 
-  text = open(filename, 'rb').read()
+  with open(filename, 'rb') as f:
+    text = f.read()
   text = convert_line_endings(text, from_eol.encode(), to_eol.encode())
-  open(filename, 'wb').write(text)
-
-# This function checks and prints out the detected line endings in the given file.
-# If the file only contains either Windows \r\n line endings or Unix \n line endings, it returns 0.
-# Otherwise, in the presence of old macOS or mixed/malformed line endings, a non-zero error code is returned.
-def check_line_endings(filename, expect_only_specific_line_endings=None, print_errors=True, print_info=False):
-  try:
-    data = open(filename, 'rb').read()
-  except Exception as e:
-    if print_errors: print("Unable to read file '" + filename + "'! " + str(e), file=sys.stderr)
-    return 1
-  if len(data) == 0:
-    if print_errors: print("Unable to read file '" + filename + "', or file was empty!", file=sys.stderr)
+  with open(filename, 'wb') as f:
+    f.write(text)
+
+
+def check_line_endings(filename, expect_only=None, print_errors=True, print_info=False):
+  """Detect inconsitent/invalid line endings.
+
+  This function checks and prints out the detected line endings in the given
+  file. If the file only contains either Windows \\r\\n line endings or Unix \\n
+  line endings, it returns 0.  Otherwise, in the presence of old macOS or
+  mixed/malformed line endings, a non-zero error code is returned.
+  """
+  if not os.path.exists(filename):
+    if print_errors:
+      print('File not found: ' + filename, file=sys.stderr)
     return 1
 
-  bad_line_ending_index = data.find(b"\r\r\n")
-  if bad_line_ending_index != -1:
+  with open(filename, 'rb') as f:
+    data = f.read()
+
+  index = data.find(b"\r\r\n")
```

Comment:
Renaming `bad_line_ending_index` to `index` looks perhaps going to a less documentative direction(?)